.github/workflows/test-runner-nightly-integration.yml

```diff
@@ -32,26 +61,38 @@ jobs:
               secret_id=${{ secrets.VAULT_APPROLE_SECRET_ID }})
           echo ::set-output name=vault-token::$VAULT_TOKEN
           echo ::add-mask::$VAULT_TOKEN
-      - name: Render configs for Test Runner
+      - name: Set permissions and render test configurations
         run: |
-          cd ${GITHUB_WORKSPACE}/${workingDir}/workspace-manager-clienttests
-          chmod +x render-config.sh
-          ./render-config.sh ${{ steps.vault-token-step.outputs.vault-token }}
+          chmod +x render_config.sh
+          ./render_config.sh ${{ steps.vault-token-step.outputs.vault-token }} dev
           chmod +r rendered
-      - name: Cache Gradle packages
-        uses: actions/cache@v2
-        with:
-          path: |
-            .gradle/caches
-            .gradle/wrapper
-          key: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle') }}
-          restore-keys: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}
-      - name: Build the WSM client and run the Basic Integration test suite
+
+      # Spin up a local server.
+      - name: Run local server (in background)
+        run: ./gradlew bootRun &
+      - name: Wait for local server to start
+        # With a timeout of 60 seconds, try to connect to localhost:8080 every 1 second.
+        run: timeout 60 bash -c 'until echo > /dev/tcp/localhost/8080; do sleep 1; done'
+
+      - name: Build the WSM client
         run: |
           echo "Building WSM client library"
           ./gradlew :workspace-manager-client:clean :workspace-manager-client:assemble
           cd workspace-manager-clienttests
           export ORG_GRADLE_PROJECT_workspacemanagerclientjar=$(find .. -type f -name "workspace-manager-client*.jar")
           echo "ORG_GRADLE_PROJECT_workspacemanagerclientjar = ${ORG_GRADLE_PROJECT_workspacemanagerclientjar}"
+
+      - name: Run the Basic Integration test suite
+        run: |
+          cd workspace-manager-clienttests
+          echo "Test Runner config"
+          ./render-config.sh ${{ steps.vault-token-step.outputs.vault-token }}
           echo "Running test suite"
           ./gradlew runTest --args="suites/BasicIntegration.json /tmp/TestRunnerResults"
+
+      - name: Upload test results to Google Bucket
+        if: ${{ always() }}
+        run: |
+          cd workspace-manager-clienttests
+          echo "Upload test results to Google Bucket"
+          ./gradlew uploadResults --args="CompressDirectoryToBucket.json /tmp/TestRunnerResults"
```

Comment:
Nit: I think CompressDirectoryToBucket is not a good name for this file because we will eventually add a BQ upload step to the same file. Perhaps something like "[UploadFor]NightlyIntegrationTests" instead?

Reply:
As presented during Perf Syncup. This was due to the caching of service account credentials (and scopes) in the Test Runner. In Test Runner library, the `StorageUtils` and `BigQueryUtils` use different scopes and are causing authentication scope permission issues when the Storage and BQ configs coexist in the same upload config file. We will address the caching issue in the Test Runner code as a separate ticket. 

I tracked this in the Google [doc](https://docs.google.com/document/d/1gYqlJTpC4C4S1K0pYikeNJx-HWNZAZJYybinmuqcSTI/edit#).